InOut/rtjack.c

```diff
@@ -1292,7 +1292,7 @@ static int32_t midi_in_open(CSOUND *csound,
       (RtJackMIDIGlobals*) csound->QueryGlobalVariableNoCheck(csound,
                                                               "_rtjackMIDIGlobals");
 
-    sprintf(clientName, "%s_in", pm->clientName);
+    snprintf(clientName, MAX_NAME_LEN+3, "%s_in", pm->clientName);
```

Comment:
The array was allocated to MAXLEN+3, so that's the length for snprintf.

I am assuming this was right and did not change it.  In the other place there is one extra character and it is +4.

Reply:
It does look like there's an inconsistency between two different versions of these fields in two different structures:

- https://github.com/csound/csound/blob/develop/H/cs_jack.h#L43
- https://github.com/csound/csound/blob/develop/InOut/rtjack.c#L1127

Based on the header file version, it feels like the intention of the `MAX_NAME_LEN` constant was the number of characters in the fields not including the terminating null.  In that case, this could potentially truncate the final "n" if the original name had the maximum length.  However, based on the jack documentation, it seems like their max length constant (returned by `jack_client_name_size()` was inclusive of the terminating null.  That being said, when we query that function and clamp to our constant, we add an extra 1:

- https://github.com/csound/csound/blob/develop/InOut/rtjack.c#L1221

So, while there's a chance this might not cause any immediate issues, it's not obvious that it's correct, either.

Reply:
Based on this, there's two ways to go:

 1) remove the +1 from the configuration variable length so that it can only go to MAX_NAME_LEN

 2) add + 1 to the jack MIDI struct and to the array/snprintf in the open device functions.

The fact that the MAX_NAME_LEN is different for the audio and midi modules is neither here nor there, so either of the two options are equally correct. I'll go for the first, which uses minimal changes.